### ISSUE TITLE
What are the available paramObjects?

### ISSUE THREAD

**Issue by ibnishak** [Arlen22/TW5Docs#1]
There are some parts where the documentation mentions "Optional hashmap of variable values to use for the rendering". For eg: [tm-download-file](http://tiddlywiki.com/#WidgetMessage%3A%20tm-download-file). What are the possible values here? If you have messed up with the core tiddlers enough, you know you can specify **filename="_myFilename_"** as a parameter for tm-download-file. But this is not documented anywhere. There could be more like this. Shouldn't there be a table of possible paramObjects for a particular topic?

**Comment by buggyj** [comment]
paramObjects are only part of the javascript api, so would be a very advanced use. They are currently currently mentioned  in the wrong places, which should probably removed.

There should be a javascript api reference somewhere, that is not part of the 'dev' documents.

**Comment by ibnishak** [comment]
> There should be a javascript api reference somewhere, that is not part of the 'dev' documents.

Is it available in dev docs now? I searched for some obvious queries which turned up nothing.

**Comment by buggyj** [comment]
@ibnishak No there is no mention anywhere, The paramObjects is just a place to locate extra message options, for present and future (new) messages -
https://github.com/Jermolene/TiddlyWiki5/commit/d93c19daaa126f0938048009a83796cb52690541

**Comment by ibnishak** [comment]
In that case your first point is all the more valid. Leaving the mentions out there simply creates confusion. Should we ask the official docs to remove it?

**Comment by buggyj** [comment]
@ibnishak the best way to ask is to do the change and  submit a pull request.

**Comment by ibnishak** [comment]
@buggyj yeah, i got to learn how to do that one of these days. Plus the apparent hesitation at the thought of audacity to make a pull request to such a project is overwhelming.  There really needs to be simpler way to suggest changes.

**Comment by buggyj** [comment]
@ibnishak, a pull request is only a suggestion.  At present the documentation is together with the source code (I think this is bad because :), only Jeremy can choose to accept changes into the project.  I think  it is less  work for him if he gets pulls request. He  can discuss the change, which can be accepted rejected or modified. It gives a solid place to talk thru changes, and others may give suggestion/help. That said,  if the change is very small I usually just open an issue because it take a bit of effort to make pull requests. 
Maybe some one else could modify the doc tiddlers and do a pull request? In any case I suggest this issue stay open until the docs are changed.
